src/AppBundle/Controller/Report/DocumentController.php

```diff
@@ -248,19 +260,70 @@ public function deleteConfirmedAction(Request $request, $documentId)
-                ? $this->generateUrl('report_documents_summary', ['reportId' => $document->getReportId()])
-                : $this->generateUrl('report_documents'        , ['reportId' => $document->getReportId()]);
+        if ($report->isSubmitted()) {
+            // if report is submitted, then this remove path has come from adding additional documents so return the user
+            // to the step 2 page.
+            $returnUrl = $this->generateUrl('report_documents', ['reportId' => $document->getReportId()]);
         } else {
-            $returnUrl = $this->generateUrl('documents_step', ['reportId' => $document->getReportId()]);
+            $reportDocumentStatus = $document->getReport()->getStatus()->getDocumentsState();
+            if (array_key_exists('nOfRecords', $reportDocumentStatus) && is_numeric($reportDocumentStatus['nOfRecords']) && $reportDocumentStatus['nOfRecords'] > 1) {
+                $returnUrl = 'summaryPage' == $request->get('from')
+                    ? $this->generateUrl('report_documents_summary', ['reportId' => $document->getReportId()])
+                    : $this->generateUrl('report_documents', ['reportId' => $document->getReportId()]);
+            } else {
+                $returnUrl = $this->generateUrl('documents_step', ['reportId' => $document->getReportId()]);
+            }
         }
 
         return $this->redirect($returnUrl);
     }
 
     /**
+     * Confirm additional documents form
+     *
+     * @Route("/report/{reportId}/documents/submit-more", name="report_documents_submit_more")
+     * @Template("AppBundle:Report/Document:submitMoreDocumentsConfirm.html.twig")
+     */
+    public function submitMoreConfirmAction(Request $request, $reportId)
+    {
+        $report = $this->getReport($reportId, self::$jmsGroups);
+
+        $fromPage = $request->get('from');
+
+        $backLink = $this->generateUrl('report_documents', ['reportId' => $reportId]);
+        $nextLink = $this->generateUrl('report_documents_submit_more_confirmed', ['reportId' => $reportId]);
+
+        return [
+            'report'   => $report,
+            'backLink' => $backLink,
+            'nextLink' => $nextLink,
+            'fromPage' => $fromPage
+        ];
+    }
+
+    /**
+     * Confirmed send additional documents.
+     *
+     * @Route("/report/{reportId}/documents/confirm-submit-more", name="report_documents_submit_more_confirmed")
+     * @Template("AppBundle:Report/Document:submitMoreDocumentsConfirmed.html.twig")
+     */
+    public function submitMoreConfirmedAction(Request $request, $reportId)
+    {
+        $report = $this->getReport($reportId, self::$jmsGroups);
+
+        // submit the report to generate the submission entry only
+        $response = $this->getRestClient()->put('report/' . $report->getId() . '/submit-documents', $report, ['submit']);
+
+        if ($response['reportId'] !== $report->getId()) {
+            throw new \RuntimeException('Unable to submit new documents');
```

Comment:
in what cases is this happening? is more likely to fail with an ecxeption from the endpoint ? if so, the previous `put(...)` will already throw an exception

Reply:
Yes this is true. Have removed. The current Id will not change since  its A PUT and already exists